File: pi.pl
sub leibniz_pi {
    my ($n) = @_;
    my $pi = 0.0;
    my $sign = 1;

    for (my $i = 0; $i < $n; $i++) {
        $pi += $sign * 1.0 / (2 * $i + 1);
        $sign = -$sign;
    }

    return 4 * $pi;
}

print leibniz_pi(2000000000), "\n";

File: pi.rs
fn leibniz_pi(n: i64) -> f64 {
    let mut pi: f64 = 0.0;
    let mut sign: f64 = 1.0;
    for i in 0..n {
        pi += sign / (2.0 * i as f64 + 1.0); 
        sign = -sign;
    }
    pi * 4.0
}

fn main() {
    let result = leibniz_pi(2000000000);
    println!("{}", result);
}
